controllers/idle_handlers.go

```diff
@@ -18,15 +18,19 @@ package controllers
 
 import (
 	"context"
+	"fmt"
 
 	ranv1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 func (r *ImageBasedUpgradeReconciler) handleAbort(ctx context.Context, ibu *ranv1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
-
-	// TODO actual steps
-	// If succeeds, return doNotRequeue
+	ostreeUndeployCommand := fmt.Sprintf("ostree admin undeploy rhcos_%s", ibu.Spec.SeedImageRef.Version)
```

Comment:
The os name can be generated with the same function and passed in to the prep script as an arg.

Reply:
currently the prep script is getting the version directly from seed image, which could be different from what is specified in the spec. What are we going to do about this?

Reply:
We probably could use utility functions for determining:
- stateroot for either side of the upgrade
- deployment index (again, for either side)
- maybe the deployment id, but I'm not sure if the LCA itself will need that

Reply:
updated